examples/visual_gen/bench_nccl_ub.py

```
@@ -0,0 +1,676 @@
+    try:
+        agent = NixlTransferAgent(name=f"bench_ring_{global_rank}", use_prog_thread=True)
+    except Exception as e:
+        return {"error": f"NixlTransferAgent init failed: {e}"}
+
+    # Exchange agent descriptors.
+    local_desc = agent.get_local_agent_desc()
+    desc_t = torch.tensor(list(local_desc), dtype=torch.uint8, device=device)
+    desc_len = torch.tensor([len(local_desc)], dtype=torch.int64, device=device)
+    all_lens = [torch.zeros(1, dtype=torch.int64, device=device) for _ in range(world_size)]
+    dist.all_gather(all_lens, desc_len, group=group)
+    max_len = int(max(int(l.item()) for l in all_lens))
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)